test/unit/client_test.coffee

```diff
@@ -107,6 +108,12 @@ describe "Client", ->
         error = _err
       return error
 
+    it "returns promise and resolves it", ->
+      promise = client.push(exception)
```

Comment:
>only incompatible browser would incur that hit yes

I honestly have very blur understanding how to load JS lib only when it is needed... Do you have a good example?